autobot-backend/api/agent_org.py

```
@@ -218,6 +223,12 @@ async def get_direct_reports(
 async def update_agent_org(
     agent_id: str,
     body: UpdateOrgRequest,
+    # Declared BEFORE the session on purpose: FastAPI resolves parameter
+    # dependencies in signature order, so the authority check runs before a
+    # database session is acquired rather than after. A gate that fires only
+    # once the handler is already holding resources is a gate in the wrong
+    # place, even when it still refuses.
+    context: TenantContext = Depends(require_reporting_line_write),
```

Ruff (0.16.3)

[warning] Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)